modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/background/task/ReindexSingleIndexerBackgroundTaskExecutor.java

```diff
@@ -27,55 +27,54 @@
 import java.util.Date;
 
 import org.osgi.framework.BundleContext;
-import org.osgi.service.component.annotations.Activate;
-import org.osgi.service.component.annotations.Component;
-import org.osgi.service.component.annotations.Deactivate;
-import org.osgi.service.component.annotations.Reference;
 
 /**
  * @author Andrew Betts
  */
-@Component(
-	property = "background.task.executor.class.name=com.liferay.portal.search.internal.background.task.ReindexSingleIndexerBackgroundTaskExecutor",
-	service = {
-		BackgroundTaskExecutor.class,
-		ReindexSingleIndexerBackgroundTaskExecutor.class
-	}
-)
 public class ReindexSingleIndexerBackgroundTaskExecutor
 	extends BaseReindexBackgroundTaskExecutor {
 
+	public ReindexSingleIndexerBackgroundTaskExecutor(
+		IndexerRegistry indexerRegistry,
+		ReindexStatusMessageSender reindexStatusMessageSender,
+		SearchEngineHelper searchEngineHelper,
+		ServiceTrackerList<Indexer<?>> systemIndexers) {
+
+		_indexerRegistry = indexerRegistry;
+		_reindexStatusMessageSender = reindexStatusMessageSender;
+		_searchEngineHelper = searchEngineHelper;
+		_systemIndexers = systemIndexers;
+	}
+
 	@Override
 	public BackgroundTaskExecutor clone() {
 		return this;
 	}
 
-	@Activate
-	protected void activate(BundleContext bundleContext) {
-		systemIndexers = ServiceTrackerListFactory.open(
-			bundleContext, (Class<Indexer<?>>)(Class<?>)Indexer.class,
-			"(system.index=true)");
+	public void closeIndexers() {
+		if (_systemIndexers != null) {
+			_systemIndexers.close();
+		}
 	}
 
-	@Deactivate
-	protected void deactivate() {
-		if (systemIndexers != null) {
-			systemIndexers.close();
-		}
+	public void openIndexers(BundleContext bundleContext) {
```

Comment:
Hey @rcsdnj, I think we can move the opening and closing logic to the `BackgroundTaskExecutorConfigurator` class and just pass the references for the trackers via constructor. What do you think?

Thx